src/gobprepare/importers/dump_importer.py

```diff
@@ -0,0 +1,134 @@
+import contextlib
+import re
+import tempfile
+from gzip import GzipFile
+from io import StringIO
+from typing import Iterator, Optional
+
+from gobconfig.datastore.config import get_datastore_config
+from gobcore.datastore.factory import DatastoreFactory
+from gobcore.datastore.objectstore import ObjectDatastore
+from gobcore.datastore.postgres import PostgresDatastore
+from gobcore.exceptions import GOBException
+from gobcore.logging.logger import logger
+
+from gobprepare.importers.typing import ReadConfig, SqlDumpImporterConfig
+
+
+@contextlib.contextmanager
+def _load_from_objectstore(datastore: ObjectDatastore) -> Iterator[str]:
+    datastore.connect()
+    try:
+        obj_info = next(datastore.query(None))
+    except StopIteration:
+        raise GOBException(f"File '{datastore.read_config['file_filter']}' not found on Objectstore.")
+    else:
+        _, obj = datastore.connection.get_object(
+            container=datastore.container_name, obj=obj_info["name"], resp_chunk_size=100_000_000
+        )
+
+    with tempfile.NamedTemporaryFile(mode="wb", suffix=".gz") as fp:
+        try:
+            for chunk in obj:
+                fp.write(chunk)
+            logger.info(f"File loaded from objectstore: {obj_info['name']}")
+            yield fp.name
+        finally:
+            datastore.disconnect()
+
+
+class SqlDumpImporter:
+    """Import a SQL dump into a PostgresDatastore table."""
+
+    def __init__(self, dst_datastore: PostgresDatastore, config: SqlDumpImporterConfig) -> None:
+        """Initialise SqlDumpImporter."""
+        self._dst_datastore = dst_datastore
+        self._encoding = config.get("encoding", "utf-8")
+        self._read_config: ReadConfig = config["read_config"]
+        self._objectstore: Optional[str] = config.get("objectstore")
+
+    def _perform_query_substitutions(self, query: str) -> str:
+        # perform the substitutions as listed in read_config.
+        if not query.endswith(self._read_config["data_delimiter_regexp"]):
+            query += ";"
+            for pattern, replacement in self._read_config["substitution"].items():
+                query = re.sub(pattern, replacement, query)
+        return query
+
+    def _instert_data(self, copy_query: str, data: str) -> None:
+        # ignore the data delimiter '\.'
+        data_to_insert = data.split(self._read_config["data_delimiter_regexp"])
+        if len(data[0]) != 0:
```

Comment:
klopt deze check? waarom eerst `data` splitsen en daarna checken of length van het 1e element != 0?

Reply:
In een van de dumps (`mks-dump_kvkveshis.sql`) staat de data leeg. Dus wel een copy query maar geen data.
```
COPY igp_pgplup_cmg_owner.kvkveshism00 (hisvesid, vestigingsnummer, kvknummer, enddate, hishibver) FROM stdin;
\.
```

Reply:
of je wil `data_to_insert` checken of je kan die split onder de `if` brengen

Reply:
Goed gezien! Fixed.